e2store/src/e2store/stream.rs

```diff
@@ -1,51 +1,59 @@
 use std::{
     fs::File,
-    io::{Read, Write},
-    path::PathBuf,
+    io::{BufReader, BufWriter, Read, Write},
+    path::Path,
 };
 
 use super::types::{Entry, Header};
 
-/// e2store/memory.rs was built to load full .era/.era2 files into memory and provide a simple API
-/// to access the data. The issue for this is for larger files this wouldn't be feasible, as the
-/// entire file would need to be loaded into memory. This is where e2store_file.rs comes in, it
-/// provides a way to read and write e2store files in a streaming fashion.
-pub struct E2StoreStream {
-    pub e2store_file: File,
+/// Streaming reader for e2store files.
+pub struct E2StoreStreamReader {
+    reader: BufReader<File>,
 }
 
-impl E2StoreStream {
-    pub fn open(e2store_path: &PathBuf) -> anyhow::Result<Self> {
-        let e2store_file = File::open(e2store_path)?;
-        Ok(Self { e2store_file })
-    }
-
-    pub fn create(e2store_path: &PathBuf) -> anyhow::Result<Self> {
-        let e2store_file = File::create(e2store_path)?;
-        Ok(Self { e2store_file })
+impl E2StoreStreamReader {
+    pub fn new(e2store_path: &Path) -> anyhow::Result<Self> {
+        let reader = BufReader::new(File::open(e2store_path)?);
+        Ok(Self { reader })
     }
 
     pub fn next_entry(&mut self) -> anyhow::Result<Entry> {
         let mut buf = vec![0; 8];
-        self.e2store_file.read_exact(&mut buf)?;
+        self.reader.read_exact(&mut buf)?;
         let header = Header::deserialize(&buf)?;
         let mut value = vec![0; header.length as usize];
-        self.e2store_file.read_exact(&mut value)?;
+        self.reader.read_exact(&mut value)?;
         Ok(Entry { header, value })
     }
+}
+
+/// Streaming writer for e2store files.
+pub struct E2StoreStreamWriter {
+    writer: BufWriter<File>,
+}
+
+impl E2StoreStreamWriter {
+    pub fn new(e2store_path: &Path) -> anyhow::Result<Self> {
```

Comment:
same here but for create even though we are using a `BufWriter` we are still creating a new file so I think that would be more clear in this context